### ISSUE TITLE
Improve the theatre extension API & UI patterns

### ISSUE THREAD

**Issue by AndrewPrifer** [theatre-js/theatre#141]
Specifically, we need:
- Toolbars that are local to a specific pane, or maybe more generally, a common set of UI patterns that extension authors can use
  Issue: if the provided patterns are not flexible enough, extension authors will revert back to home-grown UIs, defaulting the purpose of the provided UI components.
- Outlines that are local to a specific pane. When you open the r3f extension, you should only see objects that are governed by that extension. Similarly, the properties should be displayed in the context of the pane.
- Sheet objects that are ahistoric.
- Sheet objects that aren't displayed in the outline.
- A better workspace management system.

**Comment by donmccurdy** [comment]
> When you open the r3f extension, you should only see objects that are governed by that extension...

What does it mean for objects to be governed by a particular extension? I've been some trying to find the right workflow in v0.4.0 for having some objects in the 3D scene not be strictly driven by the theatre.js sheet object — e.g. the sheet object might contribute one parameter to a function that determines a 3D object's position. The 'editable' type doesn't directly set the object's position, but I'd still like to see that object in the snapshot viewport if possible. I'll write this up in a bit more detail with a new issue, but wondering if that's incompatible with the concept of "objects that are governed by that extension"?

**Comment by AndrewPrifer** [comment]
@donmccurdy no, the "only see objects that are governed by that extension" wouldn't interfere. :) This means that we should in the future probably filter our current global outline that always shows all objects to just the objects that are relevant to the current editing context. This is not super relevant if you are only using the r3f extension. Also replied to #146.

**Comment by donmccurdy** [comment]
I see, thanks! I guess my comment is not relevant in this thread. 👍

**Comment by DennisSmolek** [comment]
Following...

The scenario described by switching back to my own UI's

I needed simple sceneGraph tree views and the flow/setup in Theatre was just too much for me to figure out so I threw up some react-tree lib etc. 
At some point I thought if I have to use my own "Ui/flow to position/adjust, what am I using theatre for?"

Dont have answers yet, but  I really think theatre can be the animation tool/flow for a lot of things, I'm even dreaming up a CharacterRig/Animation controller but havent even STARTED to figure out how to make that work...
